### PR TITLE
Avoiding cannot find disk while installing with multi disk

### PR DESCRIPTION
**Pull request by fbq815**
Virtual_block: add serial for each block to make sure we could find the target stg as expected

ID: 2125185

Signed-off-by: Boqiao Fu <bfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
Test result:
JOB ID     : 79ba461bfcd45ae3f64655a5eedbbb4e66b1e70a
JOB LOG    : /root/avocado/job-results/job-2022-09-29T04.23-79ba461/job.log
 (1/5) repeat1.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: STARTED
 (1/5) repeat1.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: PASS (1077.74 s)
 (2/5) repeat2.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: STARTED
 (2/5) repeat2.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: PASS (1078.05 s)
 (3/5) repeat3.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: STARTED
 (3/5) repeat3.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: PASS (1077.23 s)
 (4/5) repeat4.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: STARTED
 (4/5) repeat4.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: PASS (1083.02 s)
 (5/5) repeat5.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: STARTED
 (5/5) repeat5.Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: PASS (1082.25 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-09-29T04.23-79ba461/results.html
JOB TIME   : 5405.44 s


@PaulYuuu @qingwangrh @maxujun @zhenyzha @menli820  please help me review this patch to see if it's working on your platform

**Comment by maxujun** [comment]
(1/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads: PASS (1187.26 s)
 (2/2) Host_RHEL.m8.u8.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads: PASS (1174.32 s)

LGTM

File: generic/tests/cfg/unattended_install.cfg [hunk 1]
```diff
@@ -50,6 +50,32 @@
             image_aio = native
         - aio_threads:
             image_aio = threads
+            blk_extra_params_stg += ",serial=TARGET_DISK0"
```
[the remark this answers is not in the document]
Comment (fbq815) [reply generic/tests/cfg/unattended_install.cfg:66]: updated

**Comment by fbq815** [comment]
Test result after update:
JOB ID     : 63297f88bceb099a69cc10de81e40ab044a45143
JOB LOG    : /root/avocado/job-results/job-2022-10-12T04.12-63297f8/job.log
 (1/1) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: STARTED
 (1/1) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.s390-virtio: PASS (1088.10 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-10-12T04.12-63297f8/results.html
JOB TIME   : 1089.29 s

@PaulYuuu @qingwangrh @zhenyzha please help me review this patch

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=unattended_install.cdrom.extra_cdrom_ks.multi_disk_install --iothread_scheme=roundrobin --nr_iothreads=2 --platform=x86_64 --guestname=RHEL.9.0.0 --driveformat=virtio_scsi --nicmodel=virtio_net --imageformat=qcow2 --machines=q35  --customsparams="vm_mem_limit = 12G\nimage_aio=native" --clone=no

(1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_native.q35: STARTED
 (1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_native.q35: PASS (1133.21 s)
 (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.q35: STARTED
 (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.q35: PASS (1128.66 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

LGTM

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
Although @menli820 tested `aio_native`, actually this change should not effects Windows guests if I know correctly. So I will merge when all platforms pass.

**Comment by zhenyzha** [comment]
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

```
 (1/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_native.arm64-pci: STARTED
 (1/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_native.arm64-pci: PASS (520.62 s)
 (2/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.arm64-pci: STARTED
 (2/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.aarch64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.arm64-pci: PASS (511.98 s)
```

**Comment by fbq815** [comment]
JOB ID     : 7dced3b1c0b19ce99f120ee5d2e916f2e9451690
JOB LOG    : /root/avocado/job-results/job-2022-10-17T01.13-7dced3b/job.log
 (1/3) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED

 (1/3) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (447.73 s)
 (2/3) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.change_media.unlimited_iothread.s390-virtio: STARTED
 (2/3) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.change_media.unlimited_iothread.s390-virtio: PASS (26.66 s)
 (3/3) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.change_media.no_iothread.s390-virtio: STARTED
 (3/3) Host_RHEL.m9.u2.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.s390x.io-github-autotest-qemu.change_media.no_iothread.s390-virtio: CANCEL: Got host qemu version:7.1.0-2, which is not in (,7.0.0-3] (1.57 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /root/avocado/job-results/job-2022-10-17T01.13-7dced3b/results.html
JOB TIME   : 480.68 s

LGTM

**Comment by fbq815** [comment]
@maxujun  please help me apply this patch on your platform

**Comment by menli820** [comment]
(1/2) Host_RHEL.m9.u0.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_native.q35: STARTED
 (1/2) Host_RHEL.m9.u0.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_native.q35: PASS (794.69 s)
 (2/2) Host_RHEL.m9.u0.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.q35: STARTED
 (2/2) Host_RHEL.m9.u0.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads.q35: PASS (832.54 s)

LGTM

**Comment by maxujun** [comment]
(1/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads: STARTED
 (1/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads: PASS (1272.11 s)
 (2/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads: STARTED
 (2/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.multi_disk_install.aio_threads: PASS (1309.02 s)

LGTM

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**